pkg/patcher/patcher.go

```diff
@@ -54,9 +55,14 @@ func PatchesPatcher(patches []string, target []byte) ([]byte, error) {
 		if strings.HasPrefix(patchString, "@") {
 			filename := patchString[1:]
 
-			contents, err = os.ReadFile(filename)
+			// Try to decrypt patch with sops first.
+			contents, err = decrypt.DecryptYamlWithSops(filename)
 			if err != nil {
-				return nil, err
+				// If it fails, read the file as is.
+				contents, err = os.ReadFile(filename)
```

Comment:
But removing that line makes test fail because of json file, which cannot be feeded into yaml decoder.

```
--- FAIL: TestPatchesPatcher (0.00s)
    patcher_test.go:115: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal array into Go value of type decrypt.sopsFile
FAIL
FAIL	github.com/budimanjojo/talhelper/pkg/patcher	0.006s
FAIL
```

Maybe that part is still needed?

Reply:
yeah really weird, I don't know what's wrong there. Maybe it's just needed then :D